src/packages/version-check/tests/version-check.test.ts

```diff
@@ -0,0 +1,737 @@
+        lastVersionLogged: "1.0.0",
+        latestVersion: "1.0.0",
+        disableInCI: false
+      };
+      vc = new VersionCheck("0.0.0", config);
+
+      assert.strictEqual(
+        vc.alreadyLoggedLatestVersion(),
+        true,
+        "alreadyLoggedLatestVersion is false when lastVersionLogged = latestVersion."
+      );
+    });
+    it("false if config.lastVersionLogged > latestVersion", () => {
+      const config = {
+        lastVersionLogged: "2.0.0",
+        latestVersion: "1.0.0",
+        disableInCI: false
+      };
+      vc = new VersionCheck("0.0.0", config);
+
+      assert.strictEqual(
+        vc.alreadyLoggedLatestVersion(),
+        true,
+        "alreadyLoggedLatestVersion is false when lastVersionLogged > latestVersion."
+      );
+    });
+  });
+
+  describe("cliMessage", () => {
+    let options;
+
+    beforeEach(() => {
+      options = {
+        upgradeType: "major",
+        packageName: "ganache",
+        currentVersion: "1.2.3",
+        latestVersion: "3.2.1",
+        disableInCI: false
+      };
+
+      vc = new VersionCheck(
+        options.currentVersion,
+        { latestVersion: options.latestVersion },
+        testLogger
+      );
+      message = "";
+    });
+
+    it("will not log if semver is the same between currentVersion and latestVersion", () => {
+      vc = new VersionCheck(
+        options.currentVersion,
+        {
+          latestVersion: options.currentVersion
+        },
+        testLogger
+      );
+
+      const didLog = !!vc.cliMessage();
+
+      assert.strictEqual(
+        didLog,
+        false,
+        "VC will log if currentVersion === latestVersion."
+      );
+    });
+
+    it("logs a single line with the currentVersion and latestVersion", () => {
+      vc = new VersionCheck(
+        options.currentVersion,
+        { latestVersion: options.latestVersion },
+        testLogger
+      );
+      const message = vc.cliMessage();
+
+      assert.strictEqual(
+        message.indexOf(options.currentVersion) >= 0,
+        true,
+        "cliMessage does not contain the currentVersion."
+      );
+      assert.strictEqual(
+        message.indexOf(options.latestVersion) >= 0,
+        true,
+        "cliMessage does not contain the latestVersion."
+      );
+    });
+
+    it("logs regardless of whether VersionCheck is enabled", () => {
+      vc.disable();
+
+      const didLog = !!vc.cliMessage();
+
+      assert.strictEqual(
+        didLog,
+        true,
+        "cliMessage, incorrectly, does not log when VC is disabled."
+      );
+    });
+    it("does not log if _currentVersion is undefined", () => {
+      vc = new VersionCheck("Bad Version");
+
+      const didLog = !!vc.cliMessage();
+
+      assert.strictEqual(didLog, false, "cliMessage logs .");
+    });
+  });
+
+  describe("log", () => {
+    it("will not log if disabled", () => {
+      vc.disable();
+
+      assert.strictEqual(message, "", "Version Check will log if disabled.");
+    });
```

Comment:
This test, and some following don't seem to have a clear confirmation of positive - ie if the thing _didn't_ happen, would it have logged?

as I read it, this test is ensuring that nothing is logged _when_ disable is called? Maybe this could be slightly clearer in the title.

ie. "will not log if canNotifyUser() is false" doesn't seem to validate anything, because no function is called that would initiate logs to be written?

Reply:
find `vc.disable` in tests